### ISSUE TITLE
pinfo/pinfo2 UX

### ISSUE THREAD

**Issue by mlucool** [jupyterlab/jupyterlab#7176]
## Description
`%pinfo` and `%pinfo2` is harder to use than in notebook. This is because in notebook, the output is output to a pane on the bottom of the screen, whereas in lab, it renders in the cell.

While it is possible to pin the output in lab, to emulate something close to notebook behavior, there are downsides that affect usability. For example, typical usage would be to bring up the doc then use it in that cell. In lab, as soon as I rerender the cell I would both not want the output pinned as well as likely still want to see that documentation on screen.

## Reproduce
```
import numpy as np
%pinfo np
```

Lab:
![image](https://user-images.githubusercontent.com/1813603/64555271-f41a2a80-d30a-11e9-870a-6172549f6dfe.png)

Notebook:
![image](https://user-images.githubusercontent.com/1813603/64555314-101dcc00-d30b-11e9-92ee-2e9cc15b716c.png)

## Context

- JupyterLab version: 1.x

**Comment by dhirschfeld** [comment]
Any reason to not just use `?`/`??`
```python
In [9]: %pinfo2?
Docstring:
Provide extra detailed information about an object.

'%pinfo2 object' is just a synonym for object?? or ??object.

In [10]: %pinfo?
Docstring:
Provide detailed information about an object.

'%pinfo object' is just a synonym for object? or ?object.
```

**Comment by mlucool** [comment]
> Any reason to not just use ?/??

No, it has the same problem as they are synonyms :). I thought that it would be easier to talk about so I used the longer name vs `?`.

**Comment by dhirschfeld** [comment]
JupyterLab also has something called [*Contextual Help*](https://github.com/jupyterlab/jupyterlab/pull/6678) which might be closer to what you're after?

**Comment by mlucool** [comment]
While its in the right direction, it is not exactly right. One problem is that it changes as you type or click on something. For example, if you type `f(g(`, the help will switch from f to g. It also is somewhat awkward to access. By default it takes up a full tab and does not remember where you left it last (e.g. the bottom of the screen).

**Comment by dhirschfeld** [comment]
Being able to pin and undock (to a normal phosphor tab) a particular help page would be a useful improvement I think.

You might get more input from actual maintainers if you cast this issue as a request for improving the contextual help rather than about `pinfo` which is ipython specific and probably not too many people are even aware of...

**Comment by goanpeca** [comment]
Now that https://github.com/jupyterlab/jupyterlab/pull/10201 is in, maybe that could be an option.
